### ISSUE TITLE
Non-Confirmable

### ISSUE THREAD

**Issue by EmileAlbert** [Tanganelli/CoAPthon#140]
Is there a way for a CoAP client to sens non-confirmable request ? I did not find a way until now, if it is possible can you tell me how  ?

Thanks in advance!

**Comment by Tanganelli** [comment]
yes, you should use client.send_request and pass as argument the coap
message. Take a look to  def send_request(self, request, callback=None,
timeout=None) in the HelperClient class.

BR

Il giorno mer 17 apr 2019 alle ore 15:26 EmileAlbert <
notifications@github.com> ha scritto:

**Comment by EmileAlbert** [comment]
First of all thanks for quik answer ! 

I see what you mean with client.send_request method, that's the method used by coapclient.py  but when I use client.put (using itself client.send_request) there is no argument for CON or NON message type and the client is waiting a response from the server. Consequently, I don't understand what I can do to change this behaviour

**Comment by Tanganelli** [comment]
you should call client.send_request, instead of client.put. Otherwise, you
have to modify the put metho to accept the type argument.

Il giorno mer 17 apr 2019 alle ore 15:59 EmileAlbert <
notifications@github.com> ha scritto:
